crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -838,6 +839,56 @@ impl RoomInfo {
     }
 }
 
+bitflags! {
+    /// Room state filter as a bitset.
+    ///
+    /// Note that [`RoomStateFilter::empty()`] doesn't filter the results and
+    /// is equivalent to [`RoomStateFilter::all()`].
+    #[derive(Clone, Copy, Debug, Eq, PartialEq)]
+    pub struct RoomStateFilter: u16 {
+        /// The room is in a joined state.
+        const JOINED   = 0b00000001;
+        /// The room is in a left state.
+        const INVITED  = 0b00000010;
+        /// The room is in a invited state.
+        const LEFT     = 0b00000100;
+    }
+}
+
+impl RoomStateFilter {
+    /// Whether the given room state matches this `RoomStateFilter`.
+    pub fn matches(&self, state: RoomState) -> bool {
+        if self.is_empty() {
+            return true;
+        }
+
+        let bit_state = match state {
+            RoomState::Joined => Self::JOINED,
+            RoomState::Left => Self::LEFT,
+            RoomState::Invited => Self::INVITED,
+        };
+
+        self.contains(bit_state)
+    }
+
+    /// Get this `RoomStateFilter` as a list of matching [`RoomState`]s.
+    pub fn as_vec(&self) -> Vec<RoomState> {
```

Comment:
Is it possible to be in a joined + left state at the same time? Or invited + joined? I reckon those states are all mutually exclusive, right?

Reply:
This is for filtering the rooms, so it is valid to request rooms that are joined and room that are invited.

One of the possible use case is for stores that can be be more performant without using the bitflag filter, e.g. in SQLite it's more efficient to request rooms with `room_state`  equal to `RoomState::Invited` or `RoomState::Joined` than to request all rooms and then filter them.